src/services/expenses.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+function findNewId() {
+  return Math.max(...expenses.map(user => user.id), 0) + 1;
+}
+
+let expenses = [];
+
+const clearExpenses = () => {
+  expenses = [];
+};
+
+const getMany = (getQuery) => {
+  const { userId, category, from, to } = getQuery;
+
+  if (!expenses.length) {
+    return [];
+  }
+
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (category) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.category === category);
+  }
+
+  if (from) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.spentAt >= from);
+  }
+
+  if (to) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.spentAt <= to);
+  }
```

Comment:
Figure out how to make the same with 1 filter. I do 4 interactions to filter expenses.